services/wallet/thirdparty/fourbyte/client.go

```diff
@@ -0,0 +1,91 @@
+package fourbyte
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/status-im/status-go/services/wallet/thirdparty"
+)
+
+type Signature struct {
+	ID   int    `json:"id"`
+	Text string `json:"text_signature"`
+}
+
+type SignatureList struct {
+	Count   int         `json:"count"`
+	Results []Signature `json:"results"`
+}
+
+type Client struct {
+	client *http.Client
+}
+
+func NewClient() *Client {
+	return &Client{client: &http.Client{Timeout: time.Minute}}
+}
+
+func (c *Client) DoQuery(url string) (*http.Response, error) {
+	resp, err := c.client.Get(url)
+
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+func (c *Client) Run(data string) (*thirdparty.DataParsed, error) {
+	methodSigData := data[2:10]
```

Comment:
Is `data` always expected to contain a `0x` prefix, and at least 8 characters afterwards? or should a validation be added to this input

Reply:
you're right I should make a check to ensure it is correctly formatted